pkg/clusteragent/externalmetrics/metrics_retriever.go

```diff
@@ -63,7 +69,31 @@ func (mr *MetricsRetriever) Run(stopCh <-chan struct{}) {
 
 func (mr *MetricsRetriever) retrieveMetricsValues() {
 	// We only update active DatadogMetrics
-	datadogMetrics := mr.store.GetFiltered(func(datadogMetric model.DatadogMetricInternal) bool { return datadogMetric.Active })
+	// We split metrics in two slices, those with errors and those without.
+	// Query first slice one by one, other as batch.
+	datadogMetrics := mr.store.GetFiltered(func(datadogMetric model.DatadogMetricInternal) bool {
+		return datadogMetric.Active && datadogMetric.Error == nil
+	})
+
+	// Do all errors warrant separate query?
+	datadogMetricsErr := mr.store.GetFiltered(func(datadogMetric model.DatadogMetricInternal) bool {
+		return datadogMetric.Active && datadogMetric.Error != nil
+	})
```

Comment:
for a futur PR. We might improve performance by creating a function that return directly 2 or 3 slices: valid, error, not use. 

maybe you can add a TODO comment about it. WDYT?

Reply:
That makes sense! I couldn't find a way for implementing it without sacrificing readability so added a comment. 

Basically we are trying to partition a slice into active and no error slice, active and error individually, and rest. A generic option is to have a func in `store` to group items by key in a map, which takes another func as an argument to map items to a key. We would have groups like 'valid', 'err_<metric_name>, 'skip' which isn't consistent key ID; mapping function wouldn't be particularly readable due to key mix IMO either. Alternative I could think of would be implementing specific logic in store which I don't really like. Let me know if I should do first option?